### DISCUSSION TITLE
LDT landcover type

### DISCUSSION THREAD

**Discussion by nchawang** [NASA-LIS/LISF#1060] [Q&A] [answered]
Hi all,

Can someone help me out with where in LDT is the code for reading in landcover files?

Also, if I have a multi year run and I would like to update the landcover files, can I regenerate the "lis_input" file from ldt every year and replace it in lis and continue the run from the restart file, will it take the updated landcover file from "lis_input"?

Thanks
Noel C.

**Comment by bmcandr** [comment, marked as answer]
Hi @nchawang,

To answer your first question, see this file https://github.com/NASA-LIS/LISF/blob/master/ldt/core/LDT_LMLCMod.F90

I don't know whether LIS has the capability you describe in your second question. @dmocko?

**Reply by dmocko** [reply]
Yes, you generally should be able to do what you are describing - use a different "lis_input" file in LIS when restarting your run.

A few caveats:

1) The landmask will need to stay the same (you can't have a land point become water, or vice versa)

2) I don't think this will work with landcover tiling, as again, the restart file assumes the tile configuration will be the same between runs.

3) Depending on what LSM you use, you may not be able to have landcover points switch from or to glacier points.  Some LSMs run very different physics over glaciers than over other landcover types.

4) Finally, the regular caveat about running with one landcover type and then abruptly switching to a new one could cause unrealistic jumps or other behavior, especially after restarting the run.

But if you want to try it anyway, just run the first year with your first "lis_input" file, stop it after a year (in this example, 31 Dec), and use these config entries in the run for the second year:

```
Start mode:                           restart
Starting year:                             [YYYY]
Starting month:                            1
Starting day:                               1
Starting hour:                             1
Starting minute:                           0
Starting second:                           0

LIS domain and parameter data file: ./lis_input.[name-of-second-year-file].nc

[LSM name] restart file:     ./LIS_RST_[LSM-name]_[YYYY]01010000.d01.nc
```

where [YYYY] = 4-digit year for the second year.  Repeat this for all years.

**Reply by nchawang** [reply]
Hi @bmcandr @dmocko 

Ah okay, Thanks for the information.
